.github/workflows/report_quiz/main.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+set -e
+
+BASE_DIR=$(dirname "${BASH_SOURCE[0]}")
+DATABASE_DIR="${BASE_DIR}/database"
+ARCHIVE_DIR="${BASE_DIR}/archive"
+TEMPFILE="$(mktemp)"
+
+# load functions
+source "${BASE_DIR}/common/utils.sh"
+source "${BASE_DIR}/quiz/functions.sh"
+source "${BASE_DIR}/leaderboard/functions.sh"
+
+# load environement variables
+source "${BASE_DIR}/env.sh"
+
+mkdir -p "${BASE_DIR}/database"
+
+if [ -z ${TELEGRAM_BOT_TOKEN} ]; then
+	echo "TELEGRAM_BOT_TOKEN is not set!"
+    exit
+elif [ -z ${TELEGRAM_CHAT_ID} ]; then
+	echo "TELEGRAM_CHAT_ID is not set!"
+    exit
+elif [ -z ${QUIZAPI_KEY} ]; then
+	echo "QUIZAPI_KEY is not set!"
+    exit
+fi
+
+# incomming parameters
+skipped_quizzes=0
+
+# determine round
+get_current_round(){
+    if [ -d ${ARCHIVE_DIR} ]; then
+        expr $(ls ${ARCHIVE_DIR} | wc -l) + 1
+    else
+	echo 1
+    fi
+}
+
+# whether there is a ongoing competition
+has_ongoing_competition() {
+    test $(ls ${DATABASE_DIR}/*.json | wc -l) -ne 0
+}
+
+# inform about a new round of quiz
+start_quiz_competition(){
+    # can't start a competition if there is a ongoing one
+    has_ongoing_competition && return 0
+
+    message="🏆 OSSCameroon Quiz Competition, Round $(get_current_round): Submissions\n\n
+
+QuizBot is now ready to accept submissions for the OSSCameroon Quiz Competition, Round.\n\n
+
+Q: How to participate\n
+A: QuizBot will send quizzes daily, you should check frequently the messages in the OSS Cameroun telegram group.
+"""
+    
+    send_message "$(escp ${message})"
+}
+
+# send a quiz
+send_quiz(){
+    while [ $skipped_quizzes -lt 3 ]; do
+        # We verify if the quiz has been proposed successfully
+        if propose_quiz > ${TEMPFILE}; then
+            # Save the quiz data in the database
+            filename=$(jq -r '.poll_id // empty' ${TEMPFILE} -r)
+            if [ ! -z "${filename}" ]; then
+                mv ${TEMPFILE} "${DATABASE_DIR}/${filename}.json"
+            fi
+            break
+        else
+            skipped_quizzes=$(expr $skipped_quizzes + 1)
+        fi
+    done
+}
+
+# update quiz data based on user answers submission
+fetch_quiz_user_answers(){
+    # require a ongoing competition
+    has_ongoing_competition || return 0
```

Comment:
We don't need that.

Reply:
I don't understand, you means `has_ongoing_competition || return 0` ?